iceoryx_posh/test/moduletests/test_popo_waitset.cpp

```diff
@@ -33,317 +34,422 @@ using namespace iox::units::duration_literals;
+{
+    EXPECT_EQ(m_sut.size(), 0);
+}
 
-TEST_F(WaitSet_test, WaitWithoutNotifyResultsInBlockingMultiThreaded)
+TEST_F(WaitSet_test, WaitSetCapacity)
 {
-    std::atomic<int> counter{0};
-    m_sut.attachCondition(m_subscriberVector.front());
-    std::thread waiter([&] {
-        EXPECT_THAT(counter, Eq(0));
-        m_syncSemaphore.post();
-        m_sut.wait();
-        EXPECT_THAT(counter, Eq(1));
-    });
-    m_syncSemaphore.wait();
-    counter++;
-    m_subscriberVector.front().notify();
-    waiter.join();
+    EXPECT_EQ(m_sut.triggerCapacity(), iox::MAX_NUMBER_OF_TRIGGERS_PER_WAITSET);
 }
 
-TEST_F(WaitSet_test, NotifyGuardConditionWhileWaitingResultsInTriggerMultiThreaded)
+TEST_F(WaitSet_test, OneAcquireTriggerIncreasesSizeByOne)
 {
-    std::atomic<int> counter{0};
-    GuardCondition guardCond;
-    m_sut.attachCondition(guardCond);
-    std::thread waiter([&] {
-        EXPECT_THAT(counter, Eq(0));
-        m_syncSemaphore.post();
-        auto fulfilledConditions = m_sut.wait();
-        EXPECT_THAT(fulfilledConditions.size(), Eq(1));
-        EXPECT_THAT(fulfilledConditions.front(), &guardCond);
-        EXPECT_THAT(counter, Eq(1));
-    });
-    m_syncSemaphore.wait();
-    counter++;
-    guardCond.trigger();
-    waiter.join();
-    m_sut.detachCondition(guardCond);
+    auto trigger1 = acquireTrigger(m_sut, 0);
+    static_cast<void>(trigger1);
+
+    EXPECT_EQ(m_sut.size(), 1);
+}
+
+TEST_F(WaitSet_test, MultipleAcquireTriggerIncreasesSizeCorrectly)
+{
+    acquireTrigger(m_sut, 5);
+    acquireTrigger(m_sut, 6);
+    acquireTrigger(m_sut, 7);
+    acquireTrigger(m_sut, 8);
+
+    EXPECT_EQ(m_sut.size(), 4);
 }
 
-TEST_F(WaitSet_test, NotifyGuardConditionOnceTimedWaitResultsInResetOfTrigger)
+TEST_F(WaitSet_test, TriggerGoesOutOfScopeReducesSize)
 {
-    GuardCondition guardCond;
-    m_sut.attachCondition(guardCond);
-    guardCond.trigger();
-    auto fulfilledConditions1 = m_sut.timedWait(1_ms);
-    EXPECT_THAT(fulfilledConditions1.size(), Eq(1));
-    EXPECT_THAT(fulfilledConditions1.front(), &guardCond);
-    guardCond.resetTrigger();
-    auto fulfilledConditions2 = m_sut.timedWait(1_ms);
-    EXPECT_THAT(fulfilledConditions2.size(), Eq(0));
-    m_sut.detachCondition(guardCond);
+    acquireTrigger(m_sut, 1);
+    acquireTrigger(m_sut, 2);
+    {
+        auto trigger3 = acquireTrigger(m_sut, 3);
+        auto trigger4 = acquireTrigger(m_sut, 4);
+        removeTrigger(trigger3->value().getUniqueId());
+        removeTrigger(trigger4->value().getUniqueId());
+    }
+
+    EXPECT_EQ(m_sut.size(), 2);
+}
+
+TEST_F(WaitSet_test, MovingAssignTriggerReducesSize)
+{
+    auto trigger1 = acquireTrigger(m_sut, 0);
+    TriggerHandle trigger2;
+    trigger2 = std::move(trigger1->value());
+
+    EXPECT_EQ(m_sut.size(), 1);
+}
+
+TEST_F(WaitSet_test, MoveCTorTriggerDoesNotChangeSize)
+{
+    auto trigger1 = acquireTrigger(m_sut, 0);
+    auto trigger2(std::move(*trigger1));
+
+    EXPECT_EQ(m_sut.size(), 1);
 }
```

Comment:
Maybe I missed it, but there seems to be a test missing that checks whether timed wait actually returns after the wait times out, i.e. is non-blocking.
 This cannot be exact though, but maybe waiting for two times the stated period of say 100ms would make for a reasonably robust test (in theory with sufficient load this still may not be enough....).

Reply:
The test `TimedWaitReturnsNothingWhenNothingTriggered` is testing this.